examples/ble_passive_scanning/advertisement.h

```diff
@@ -2,31 +2,41 @@
 
 #include <cstring>
 
-// this is left outside the class because it doesn't work
-// code-generation bug?!
-const int ADDRESS_SIZE = 6;
-const int DATA_SIZE = 32;
-const int DATA_START = 8;
-const int DATA_END = 39;
-const int ADV_TYPE = 0;
+const unsigned char HEADER_START = 0;
+const unsigned char HEADER_SIZE = 2;
+const unsigned char ADDRESS_START = 2;
+const unsigned char ADDRESS_SIZE = 6;
+const unsigned char DATA_START = 8;
+const unsigned char DATA_MAX_SIZE = 31;
+
+const unsigned char ADV_MIN_SIZE = 8;
+const unsigned char ADV_MAX_SIZE = HEADER_SIZE + ADDRESS_SIZE + DATA_MAX_SIZE;
+
+const unsigned char PDU_ADDRESS_HEADER_MASK = 0x3F;
```

Comment:
I don't understand this, is it meant to get rid of the 2 RFU bits in the second byte of the header?

IIRC, the actual length in the header includes the address which is 6-37 bytes.

Reply:
I used a reference from texas instruments: http://www.ti.com/lit/an/swra475a/swra475a.pdf

PDU-Length is between 6 and 37 bytes including address. This leads to a data part between 0 and 31 bytes. My reference says that the PDULength field in the header has a length of 6 bits. Makes sense to me as the maximum of 37 perfectly fits into 6 bits.

-> I mask the length of the field.

Did I get something wrong here?

Reply:
No, it's correct but I was more referring to the name of constant. `PDU_ADDRESS_HEADER_MASK` is not clear to me I would suggest `PDU_LEN_HEADER_MASK` or something similar.

Reply:
Ah now I see your point. You are of course totally right.